### DISCUSSION TITLE
How Big Is a Privacy Budget ?

### DISCUSSION THREAD

**Discussion by lilasaba** [OpenMined/courses#240] [Course: Our Privacy Opportunity] [answered]
My question came up while watching `Lesson 5 / Consumers Need Structured Transparency Institutions`.

In a future scenario it might become possible that we can track privacy spending at a database or individual level with an *automated privacy budgeting* tool. Now, I clearly do not understand the details here, but from the discussion it seemed that this budget is very finite.
For example, if we have a medical database, after a couple hundred queries, the budget might run out. So after that, every additional query increases the risk of sensitive info is being released.
As far as I understand now the budget/epsilon also depends on the database size and nr of variables, but rather sooner than later the database just cannot be used anymore.

So, what are the ratios (query per database) roughly, that the current DP technologies support ?

Thank you !

**Comment by hershd23** [comment, marked as answer]
Hi @lilasaba that is a good question.
I believe this [piece](https://aircloak.com/explaining-differential-privacy/) should have the answer to your question. 

Moreover the privacy budgeting tools like the one you are talking about have already been developed. One such tool is Harvard's [PSI](http://psiprivacy.org/static/about/index.html).

I am also linking a paper that I read recently and found to be pretty cool. It proposes a cost model that links privacy to monetary budget. Such models can be used in the future to select optimal epsilon budgets according to our requirements
https://arxiv.org/pdf/2003.00973.pdf

Hope these resources answer your question

**Reply by lilasaba** [reply]
Thanks, yes they do !

> Setting epsilon to 0.5, we could build user-count histograms of 3 columns and take the mean of two more. The tool estimated the 95% error on the mean to be 3%, and on the counts to be ±60 (around 5% to 10% for most of the histogram bars). These are reasonable and useful answers, but those 5 queries exhausted the budget.

Well, not the best outlook, but at least a start :)
